volplugin/runtime.go

```diff
@@ -1,96 +1,52 @@
 package volplugin
 
 import (
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
-	"net/http"
-	"reflect"
-	"time"
 
 	"github.com/contiv/errored"
 	"github.com/contiv/volplugin/config"
-	"github.com/contiv/volplugin/storage"
+	"github.com/contiv/volplugin/errors"
+	"github.com/contiv/volplugin/watch"
 
 	log "github.com/Sirupsen/logrus"
 )
 
-func (dc *DaemonConfig) pollRuntime(volumeName string, mc *storage.Mount) {
+func (dc *DaemonConfig) pollRuntime() {
+	volumeChan := make(chan *watch.Watch)
+	dc.Client.WatchVolumeRuntimes(volumeChan)
 	for {
-		dc.runtimeMutex.RLock()
-		stop := dc.runtimeStopChans[volumeName]
-		dc.runtimeMutex.RUnlock()
-		select {
-		case <-stop:
-			return
-		case <-time.After(dc.Global.TTL):
-			log.Debugf("Checking runtime parameters for volume %q (ttl: %d)", volumeName, dc.Global.TTL)
+		volWatch := <-volumeChan
 
-			log.Debugf("Requesting runtime configuration for volume %q", volumeName)
-			resp, err := http.Get(fmt.Sprintf("http://%s/runtime/%s", dc.Master, volumeName))
-			if err != nil {
-				log.Error(errored.Errorf("Error processing runtime update").Combine(err))
-				continue
-			}
+		fmt.Printf("%#v\n", volWatch)
```

Comment:
No. :D

On 27 Jun 2016, at 13:27, unclejack wrote:

>>
>> -			log.Debugf("Requesting runtime configuration for volume %q", 
>> volumeName)
>> -			resp, err := http.Get(fmt.Sprintf("http://%s/runtime/%s", 
>> dc.Master, volumeName))
>> -			if err != nil {
>> -				log.Error(errored.Errorf("Error processing runtime 
>> update").Combine(err))
>> -				continue
>> -			}
>> +		fmt.Printf("%#v\n", volWatch)
>
> Should this be in here?
>
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/volplugin/pull/313/files/a4dd7184906418e9c3fa0abbbfa0cad298fd36f4#r68649408